### PR TITLE
[FastPR][Core] ModelPartIO geometries submodelparts fix

### PR DESCRIPTION
**Pull request by rubenzorrilla**
**📝 Description**
Hotfix to also do the parsing of the submodelpart geometries.

@jginternational this should fix the issue you reported to us yesterday.

### CODE REVIEW

File: kratos/sources/model_part_io.cpp [hunk 1]
```diff
@@ -3637,6 +3639,29 @@ void  ModelPartIO::ReadSubModelPartConditionsBlock(ModelPart& rMainModelPart, Mo
...
+    while (!mpStream->eof())
+    {
+        ReadWord(word); // Reading the node id or End
+        if (CheckEndBlock("SubModelPartGeometries", word))
+            break;
+
+        ExtractValue(word, geometry_id);
+                    ordered_ids.push_back(geometry_id);
+    }
+    std::sort(ordered_ids.begin(), ordered_ids.end());
```
Comment (philbucher) [inline kratos/sources/model_part_io.cpp:3661]: Why the sort?

This makes it inconsistent with the mdpa file

Or is this also done for elems and conds?

File: kratos/sources/model_part_io.cpp [hunk 2]
```diff
@@ -3637,6 +3639,29 @@ void  ModelPartIO::ReadSubModelPartConditionsBlock(ModelPart& rMainModelPart, Mo
...
+{
+    KRATOS_TRY
+
+    SizeType geometry_id;
+    std::string word;
+    std::vector<SizeType> ordered_ids;
+
+    while (!mpStream->eof())
+    {
+        ReadWord(word); // Reading the node id or End
```
Comment (philbucher) [inline kratos/sources/model_part_io.cpp]: Wrong comment

**Review by philbucher — COMMENTED** [review]
Minor otherwise looks good

File: kratos/sources/model_part_io.cpp (hunk 1 above)
Comment (rubenzorrilla) [reply kratos/sources/model_part_io.cpp:3661]: Basically I took what we do for elements and conditions and I applied here as well.

File: kratos/sources/model_part_io.cpp (hunk 2 above)
Comment (rubenzorrilla) [reply kratos/sources/model_part_io.cpp]: Copy & paste error. Thanks for spotting.

File: kratos/sources/model_part_io.cpp (hunk 1 above)
Comment (rubenzorrilla) [reply kratos/sources/model_part_io.cpp:3661]: I've been checking the `AddGeometries` of the `ModelPart` and I don't see why the `std::sort` is needed. As far as I understand it shouldn't affect the performance of the key (id) search that we do in there as the underlying geometry container is an `std::unordered_map`.

**Review by philbucher — APPROVED** [review]

File: kratos/sources/model_part_io.cpp (hunk 1 above)
Comment (philbucher) [reply kratos/sources/model_part_io.cpp:3661]: Seems that you didn't remove it but up to you

Comment (rubenzorrilla) [reply kratos/sources/model_part_io.cpp:3661]: Yes, I want to confirm the removal with the @KratosMultiphysics/technical-committee guys.

**Comment by rubenzorrilla** [comment]
For the record. We merge the hotfix as it is and we'll discuss later on about the sorting.

**Pull request merged**